doc/adr002-storage.md

```diff
@@ -0,0 +1,58 @@
+# ADR002: Storage Layout
+
+**Author:** Igor Konnov, 2024
+
+This ADR discusses the simple layout of the Soroban transactions that are
+fetched from a Stellar network.
+
+Our goals are:
+
+ - Store contract calls retrieved from the Horizon API, see [storage.ts][].
+ - Access calls indendently for verification.
+ - Tag calls as `stored`, `verified`, `failed`, etc.
+
+To keep things simple in the activation stage, we are simply using the
+filesystem instead of a fully-featured database. This is sufficient to
+demonstrate our idea.  When the users expect a storage that its durable,
+consistent, fail-free, etc., we would have to implement the storage with a solid
+database library.
+
+In the rest of this document, we refer to the root of the storage directory as
+`${STOR}$`. We expect the user to provide the Solarkraft commands with the
+root directory. By default, `${STOR}` would point to
+`$HOME/.solarkraft/.stor`.
+
+## 1. Storing extracted calls
+
+We have the following requirements to the storage:
+
+ - It should support contract calls (from transactions) that are extracted for
+   different contracts.
+ 
+ - It should be relatively easy to order these transactions, e.g., by ledger height.
+
+ - Transactions that happen in the same block should not collide.
+
+
+Given all these requirements, we store every contract call in its own file called:
+
+```
+${STOR}/${contractId}/${height}/entry-${txHash}.json
```

Comment:
So you want to say anything about the content format of these files? 

Even if it's just pointing to the implementation, I think it would be helpful.

Reply:
added a comment